pilot/model/config.go

```diff
@@ -386,6 +398,23 @@ func ResolveHostname(meta ConfigMeta, svc *routing.IstioService) string {
 	return out
 }
 
+// TODO: support IP addresses (?)
+// TODO: support non-FQDN names in Consul/Eureka
+// support 3 forms
+// reviews => reviews.default.svc.cluster.local
+// reviews.sales => reviews.sales.svc.cluster.local
+// reviews.default.svc.cluster.local => reviews.default.svc.cluster.local (no change)
+func ResolveFQDN(meta ConfigMeta, name string) string {
+	switch strings.Count(name, ".") {
+	case 0:
+		return fmt.Sprintf("%s.%s.svc.%s", name, meta.Namespace, meta.Domain)
```

Comment:
We don't need this precomputation anymore (more like we can't). We should probably have a platform's domain as part of service model or something (e.g., service.consul or svc.cluster.local). You resolve the namespace based on the service (source) where the route rule is applied. So, its destinationname.source-namespace.... or its just destinationname.service.consul ...

Reply:
Could you elaborate on this? I am not sure what you mean.

Reply:
Can we plumb the platform type into the ResolveFQDN function? So that if its Kube, we can use Namespace and svc.cluster.local from the rule's metadata field.. if its Consul, we use service.consul. If its Eureka, don't do anything. If its CloudFoundry, dont do anything..

Reply:
Because, I would like to use a simple service name in Consul environment as well.

Reply:
Alternatively, you could tweak this logic to check if namespace is emtpy.. and if so, do name+domain.. if its not emtpy, do, name+namespace+domain